### ISSUE TITLE
dotnet8 install via winetricks not recognized by application

### ISSUE THREAD

**Issue by gc-user** [Winetricks/winetricks#2276]
I installed dotnet8 via winetricks, which then installed dotnet8 for x86 and x64. But when starting the program that asked for dotnet8 in the first place, it still asks for it. Uninstalling and reinstalling dotnet8 (via winetricks) didn't change anything.

Only when I downloaded and installed dotnet8 x64 manually would the program recognize dotnet8 as installed and start and run as expected.

I noticed that the cached install exe files for dotnet8 in the winetricks cache folder are about half the size as the one I downloaded manually. Also, winetricks complained when calling dotnet8:

```

warning: Unknown file arch of /usr/bin/wineserver.
 
warning: Unknown file arch of /usr/bin/wine.

warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
 
warning: WoW64 type could not be detected.

```

I've seen at least some of those warnings before in other contexts, never cared because everything seems to work ok otherwise. But maybe this can be a factor for why dotnet8 installed by winetricks is not recognized by the program that asked for it.

Other dotnet versions never had such a problem, i.e. they all worked fine when installing via winetricks, namely dotnet20, dotnet40, dotnet45, dotnet46, dotnet461, dotnet462, dotnet472.

![ditnet8 installer file sizes](https://github.com/user-attachments/assets/1b2a4f23-f64e-4ad9-9a4d-004671b75714)

**Comment by ThePythonDev** [comment]
Strange... Is it errors or warnings?

**Comment by gc-user** [comment]
Just the ones I mentioned.
And of course the ones of the application that needs dotnet8 complaining about dontnet8 missing/ not being installed.

**Comment by austin987** [comment]
The sizes look as expected. Note that desktop version is different from the runtime.

Note that newer dotnet versions (i.e., 5.0+) require the exact version that the application was compiled against. I.e., 8.0.0 won't work for an application expecting 8.0.2.

Those warnings are concerning, but probably not the problem here (and should be their own issue).

**Comment by gc-user** [comment]
Thanks for the input!
Good to know about the version. But as the version (8.08) according to the names of the downloaded files, seems to be the same, shouldn't it work with both, the installer downloaded by winetricks and downloaded manually - if it works with the latter?

Also, the manually downloaded installer file has "desktop" and "runtime" in the name, so I guess, "desktop" trumps "runtime" in regards to determining which one it is.

**Comment by austin987** [comment]
> Thanks for the input! Good to know about the version. But as the version (8.08) according to the names of the downloaded files, seems to be the same, shouldn't it work with both, the installer downloaded by winetricks and downloaded manually - if it works with the latter?

Yes, it should.

> Also, the manually downloaded installer file has "desktop" and "runtime" in the name, so I guess, "desktop" trumps "runtime" in regards to determining which one it is.

Desktop contains the runtime, as well as GUI components. Runtime is for CLI applications.

What application are you testing with? Does it have a a free download available?

**Comment by gc-user** [comment]
> Desktop contains the runtime, as well as GUI components. Runtime is for CLI applications.
I think I got it now! Thanks!

> What application are you testing with? Does it have a a free download available?

It is Disk Drill (https://www.cleverfiles.com/data-recovery-software.html). I tested it again, un- and re-installing the x64 version of .NET v8.0.8 as plain Runtime edition and as Desktop Runtime edition manually - via the respective downloaded installer file (Runtime downloaded by winetricks, Desktop downloaded manually by myself). The program requires 8.0.0 (mind., I guess). It only runs with Desktop Runtime. Plain Runtime seems to be not enough.

Not sure though, if winetricks would even be able to download the Desktop Runtime edition. I guess, it always opts for the plain Runtime edition.

**Comment by austin987** [comment]
> > What application are you testing with? Does it have a a free download available?
> 
> It is Disk Drill (https://www.cleverfiles.com/data-recovery-software.html).

What wine version are you using? It doesn't install here with 9.19 or 9.0.

> I tested it again, un- and re-installing the x64 version of .NET v8.0.8 as plain Runtime edition and as Desktop Runtime edition manually.

Please use a clean WINEPREFIX when reporting/testing issues.

> Not sure though, if winetricks would even be able to download the Desktop Runtime edition. I guess, it always opts for the plain Runtime edition.

`winetricks dotnetdesktop8` will install the desktop runtime.

**Comment by gc-user** [comment]
> What wine version are you using? It doesn't install here with 9.19 or 9.0.
9.11 because every version after that makes one of the programs I have installed show a bug.
But actually, I didn't even install Disk Drill. Whenever possible I use Universal Extractor to extract installer files and put the resulting folder into a special path, essentially a drive_d for "portable installations". I started that back when I was still using Windows to save time and headaches with my then backup/update regime and to have less / no leftovers after possible uninstall. And this works in the case of Disk Drill, two. On wine I use this procedure for different reasons, one being not having leftover files after "uninstall" when it turns out a program doesn't work with wine.

But in this case maybe the install would have prompted to install "dotnetdesktop8" (the installer, not winetricks) or even downloaded it automatically. Which wouldn't have made a difference though.
 
> Please use a clean WINEPREFIX when reporting/testing issues.
I'll try. But setting up a wineprefix from scratch is a pita, because all the "dependencies" (like all the dotnets) have to be installed that were installed over time and maybe settings changed which I can't even remember most of the time why and how I did something to make some Windows program work with wine.


> `winetricks dotnetdesktop8` will install the desktop runtime.
Thanks for the info! Good to know! I guess that answers my initial question / report, as the error message of that program states that .NET Desktop Runtime 8.0.0 x64 is missing. Now that I know the difference between Desktop Runtime and Runtime, and that winetricks can also download / install the former, I see that the error lies with me, not winetricks. :-)

Thanks and sorry for the confusion I caused!
